lib/galaxy/schema/library_contents.py

```diff
@@ -0,0 +1,329 @@
+
+
+class LibraryContentsShowFolderResponse(LibraryContentsShowResponse):
+    model_class: Annotated[Literal["LibraryFolder"], ModelClassField(Literal["LibraryFolder"])]
+    id: EncodedLibraryFolderDatabaseIdField
+    parent_id: Optional[EncodedLibraryFolderDatabaseIdField]
+    description: str
+    item_count: int
+    deleted: bool
+    library_path: List[str]
+
+
+class LibraryContentsShowDatasetResponse(LibraryContentsShowResponse):
+    model_class: Annotated[Literal["LibraryDataset"], ModelClassField(Literal["LibraryDataset"])]
+    id: EncodedDatabaseIdField
+    ldda_id: EncodedDatabaseIdField
+    folder_id: EncodedLibraryFolderDatabaseIdField
+    state: str
+    file_name: str
+    created_from_basename: Optional[str]
+    uploaded_by: Optional[str]
+    message: Optional[str]
+    date_uploaded: str
+    file_size: int
+    file_ext: str
+    data_type: str
+    misc_info: Optional[str]
+    misc_blurb: Optional[str]
+    peek: Optional[str]
+    uuid: str
+    tags: TagCollection
+
+    # metadata fields
+    model_config = ConfigDict(extra="allow")
+
+
+class LibraryContentsCreateResponse(Model):
+    name: str
+    url: str
+
+
+class LibraryContentsCreateFolderResponse(LibraryContentsCreateResponse):
+    id: EncodedLibraryFolderDatabaseIdField
+
+
+class LibraryContentsCreateFileResponse(LibraryContentsCreateResponse):
+    id: EncodedDatabaseIdField
+
+
+class LibraryContentsCreateFolderListResponse(RootModel):
+    root: List[LibraryContentsCreateFolderResponse]
+
+
+class LibraryContentsCreateFileListResponse(RootModel):
+    root: List[LibraryContentsCreateFileResponse]
+
+
+class LibraryContentsCreateDatasetResponse(Model):
+    # id, library_dataset_id, parent_library_id should change to EncodedDatabaseIdField latter
+    # because they are encoded ids in _copy_hda_to_library_folder and _copy_hdca_to_library_folder
+    # functions that are shared by LibraryFolderContentsService too
+    id: str
+    hda_ldda: str
+    model_class: Annotated[
+        Literal["LibraryDatasetDatasetAssociation"], ModelClassField(Literal["LibraryDatasetDatasetAssociation"])
+    ]
+    name: str
+    deleted: bool
+    visible: bool
+    state: str
+    library_dataset_id: str
+    file_size: int
+    file_name: str
+    update_time: str
+    file_ext: str
+    data_type: str
+    genome_build: str
+    misc_info: Optional[str]
+    misc_blurb: Optional[str]
+    created_from_basename: Optional[str]
+    uuid: str
+    parent_library_id: str
+
+    # metadata fields
+    model_config = ConfigDict(extra="allow")
+
+
+class LibraryContentsCreateDatasetCollectionResponse(RootModel):
+    root: List[LibraryContentsCreateDatasetResponse]
+
+
+class LibraryContentsDeleteResponse(Model):
+    id: EncodedDatabaseIdField
+    deleted: bool
+
+
+class LibraryContentsPurgedResponse(LibraryContentsDeleteResponse):
+    purged: bool
+
+
+AnyLibraryId = Union[DecodedDatabaseIdField, LibraryFolderDatabaseIdField]
```

Comment:
I'm a bit confused by this. I thought the library ID will **always** be a `DecodedDatabaseIdField` and **never** a `LibraryFolderDatabaseIdField` (that would be the `root_folder_id`)

Reply:
Well, I had to include folder_id because there were tests that used this:
https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy_test/api/test_libraries.py#L608
Maybe the test here is wrong and it should be library id instead of folder id here?

Reply:
It seems that it should be the problem from the function because it doesn't make sense to use folder_id instead of library_id and it wasn't checked in the old API because it doesn't need it to create a dataset in the library.